src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -4082,16 +4083,29 @@ void CodeGen::genCodeForShift(GenTree* tree)
         }
         else
         {
-            // First, move the operand to the destination register and
-            // later on perform the shift in-place.
-            // (LSRA will try to avoid this situation through preferencing.)
-            if (tree->GetRegNum() != operandReg)
+            int typeWidth    = genTypeSize(genActualType(targetType)) * 8;
+            int shiftByValue = (int)shiftBy->AsIntConCommon()->IconValue();
+
+            // Try to emit rorx if BMI is available instead of rol
+            if (compiler->compOpportunisticallyDependsOn(InstructionSet_BMI2) && tree->OperIs(GT_ROL) &&
```

Comment:
Do we not want a similar path for `GT_ROR`?

Reply:
Ah, indeed. Both `BitOperations.RotateRight` and `BitOperations.RotateLeft` don't emit it but looks like `fgRecognizeAndMorphBitwiseRotation` still can emit it

Reply:
Does `RotateRight` get recognized as a ROL instead? I wonder what causes that 🤔

Reply:
just checked, `BitOperations.RotateRight` is correctly recognized as `ror` when offset is not a const. but when it's a const - [this ](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/morph.cpp#L14998-L15002)condition is false (because that ADD is folded).